src/tools/perf/lib/libperf.c

```diff
@@ -1620,15 +1620,20 @@ static ucs_status_t ucp_perf_setup(ucx_perf_context_t *perf)
     unsigned i, thread_count;
     size_t message_size;
 
-    ucp_params.field_mask        = UCP_PARAM_FIELD_FEATURES;
-
     ucp_params.field_mask   = UCP_PARAM_FIELD_FEATURES |
                               UCP_PARAM_FIELD_REQUEST_SIZE |
                               UCP_PARAM_FIELD_REQUEST_INIT;
     ucp_params.features     = 0;
     ucp_params.request_size = sizeof(ucp_perf_request_t);
     ucp_params.request_init = ucp_perf_request_init;
 
+    if (perf->params.thread_count > 1) {
+        /* when there is more than one thread, a ucp_worker would be created for
+         * each. all of them will share the same ucp_context */
+        ucp_params.features          |= UCP_PARAM_FIELD_MT_WORKERS_SHARED;
```

Comment:
```
ucp_params.field_mask |= UCP_PARAM_FIELD_MT_WORKERS_SHARED;
```

Reply:
yeah, this issue was introduced before, will fix in another PR

Reply:
@dmitrygx Thanks for catching this! Please review PR https://github.com/openucx/ucx/pull/5015
@yosefe I'll rebase on top of this PR of course.

Reply:
> @dmitrygx Thanks for catching this! Please review PR #5015

@alinask thanks! done